### PR TITLE
Bug 2011368: fix PipelineRun visualization tooltip to show duration

### PR DESCRIPTION
**Pull request by debsmita1**
**Fixes:**
https://bugzilla.redhat.com/show_bug.cgi?id=2011368

**Root analysis:**
Tooltip for tasks in pipeline visualization shows when the task ran instead of how long it ran.

**Solution description:**
- pipeline run duration now shows how long the tasks ran

**Screenshot:**
Screenshot from 2022-01-13 18-33-56
Screenshot from 2022-01-10 17-10-10
Screenshot from 2022-01-10 17-04-46

### CODE REVIEW

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/pipeline-metrics-utils.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ import {
   dateFormatterNoYear,
   parsePrometheusDuration,
 } from '@console/internal/components/utils/datetime';
+import i18n from '@console/internal/i18n';
```
Comment (jerolimov) [inline frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/pipeline-metrics-utils.ts:13]: There is already an import `import i18next from 'i18next';`, all other translation code here uses `i18next.t(`, could we reuse this here as well?

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/pipeline-metrics-utils.ts [hunk 2]
```diff
@@ -119,13 +120,43 @@ export const getXaxisValues = (timespan: number): number[] => {
   return xValues.slice(0, numDays);
 };
 
-export const getYaxisValues = (seconds: number): string => {
-  if (seconds < 60) {
-    return `${seconds}s`;
+export const getDuration = (seconds: number, list?: boolean): string => {
```
Comment (jerolimov) [inline frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-metrics/pipeline-metrics-utils.ts:123]: I would suggest a name that is independent of the use case "list" or "detail page" and use a more generic term like "short" or "long" here. Wdyt?

```suggestion
export const getDuration = (seconds: number, long: boolean): string => {
```

File: frontend/packages/pipelines-plugin/src/components/pipelines/detail-page-tabs/pipeline-details/pipeline-step-utils.ts [hunk 3]
```diff
@@ -35,10 +35,7 @@ const getMatchingStepDuration = (matchingStep?: TaskStatusStep) => {
   if (!matchingStep) return '';
 
   if (matchingStep.terminated) {
-    return calculateRelativeTime(matchingStep.terminated.startedAt);
-  }
-  if (matchingStep.running) {
-    return calculateRelativeTime(matchingStep.running.startedAt);
+    return calculateDuration(matchingStep.terminated.startedAt, matchingStep.terminated.finishedAt);
   }
```
Comment (jerolimov) [inline frontend/packages/pipelines-plugin/src/components/pipelines/detail-page-tabs/pipeline-details/pipeline-step-utils.ts:39]: We should also show a duration if the pipeline step is running. This gives the user a creat indicator if its need 10%, 50% or 90% complete.

```suggestion
  if (matchingStep.terminated) {
    return calculateDuration(matchingStep.terminated.startedAt, matchingStep.terminated.finishedAt);
  }
  if (matchingStep.running) {
    return calculateDuration(matchingStep.running.startedAt, new Date() /* or make the second parameter optional and do this in calcDuration? */);
  }
```

File: frontend/packages/pipelines-plugin/src/utils/pipeline-utils.ts [hunk 4]
```diff
@@ -313,28 +314,11 @@ export const getPipelineRunWorkspaces = (
   );
 };
 
-export const calculateRelativeTime = (startTime: string, completionTime?: string) => {
+export const calculateDuration = (startTime: string, completionTime: string, list?: boolean) => {
```
Comment (jerolimov) [inline frontend/packages/pipelines-plugin/src/utils/pipeline-utils.ts:317]: To calculate a duration without an end date it's maybe easier to switch to Date here and then parse the strings in calling functions? But that's not required.

```suggestion
export const calculateDuration = (startTime: Date, endTime: Date, long: boolean) => {
```

**Review by jerolimov — CHANGES_REQUESTED** [review]
Hey, nice PR. :+1:

Unluckily the [Intl.RelativeTimeFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/RelativeTimeFormat) doesn't match our requirements here.

Also if we have a translation for the "hour(s)" it's not perfect, but I think the old version was not translated at all. So this is also a win here.

But I would suggest showing the duration also if the step or pipeline is still running ("running since"). Everything else is fine from my side.

**Review by jerolimov — COMMENTED** [review]
Looks good to me! :+1: Would like that @karthikjeeyar to add lgtm as well. :)

/approve

**Review by jerolimov — APPROVED** [review]

**Comment by karthikjeeyar** [comment]
Thanks @debsmita1 for this PR, duration field now shows correct representation of the data.
/lgtm

**Pull request merged**